solidity_contracts/src/CairoPrecompiles/CairoLib.sol

```diff
@@ -0,0 +1,216 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.7.0 <0.9.0;
+
+library CairoLib {
+    /// @dev The Cairo precompile contract's address.
+    address constant CAIRO_PRECOMPILE_ADDRESS = 0x0000000000000000000000000000000000075001;
+
+    /// @notice Performs a low-level call to a Cairo contract deployed on the Starknet appchain.
+    /// @dev Used with intent to modify the state of the Cairo contract.
+    /// @param functionSelector The function selector of the Cairo contract function to be called.
+    /// @param data The input data for the Cairo contract function.
+    /// @return returnData The return data from the Cairo contract function.
+    function callContract(
+        uint256 contractAddress,
+        uint256 functionSelector,
+        uint256[] memory data
+    ) internal returns (bytes memory returnData) {
+        bytes memory callData = abi.encodeWithSignature(
+            "call_contract(uint256,uint256,uint256[])",
+            contractAddress,
+            functionSelector,
+            data
+        );
+
+        (bool success, bytes memory result) = CAIRO_PRECOMPILE_ADDRESS.call(callData);
+        require(success, "CairoLib: call_contract failed");
+
+        returnData = result;
```

Comment:
if you name the returned value, you can skip this and use the same name directly. I my self prefer not to name it and use the `return` statement, more explicit imo